File: Deepseek-Coder-33b-Instruct/stage2/host_data_construct_if_clause.c
#include <stdlib.h>
#include <stdio.h>
#include <openacc.h>

#define N 100

int main() {
    int a[N], b[N];
    int i;

    // Initialize arrays
    for (i = 0; i < N; i++) {
        a[i] = i;
        b[i] = 0;
    }

    // Copy data from host to device
    #pragma acc data copyin(a) copyout(b)
    {
        // Use the host_data construct with an if clause
        #pragma acc host_data if(1)
        {
            for (i = 0; i < N; i++) {
                b[i] = a[i];
            }
        }
    }

    // Check the results
    for (i = 0; i < N; i++) {
        if (b[i] != a[i]) {
            printf("Error: b[%d] = %d, expected %d\n", i, b[i], a[i]);
            return 1; // Return non-zero to indicate failure
        }
    }

    printf("Success!\n");
    return 0; // Return zero to indicate success
}

File: Deepseek-Coder-33b-Instruct/stage2/host_data_construct_if_clause.f90
#ifndef T1
!T1:host_data construct if clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER :: a, b
        LOGICAL :: condition

        ! Initialize the condition
        condition = .FALSE.

        !$acc host_data if(condition)
        a = 1
        !$acc end host_data

        !$acc host_data if(.not. condition)
        b = 2
        !$acc end host_data

        ! Check if the values of a and b are as expected
        IF (a .ne. 1) THEN
          errors = errors + 1
        END IF
        IF (b .ne. 2) THEN
          errors = errors + 1
        END IF

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM